Sources/AppcuesKit/Presentation/ExperienceRendering/ExperienceRenderer.swift

```diff
@@ -57,7 +152,7 @@ internal class ExperienceRenderer: ExperienceRendering {
         // only track analytics on published experiences (not previews)
         // and only add the observer if the state machine is idling, otherwise there's already another experience in-flight
         if published && stateMachine.state == .idling {
-            self.stateMachine.addObserver(analyticsObserver)
+            stateMachine.addObserver(ExperienceStateMachine.AnalyticsObserver(container: container))
```

Comment:
I think if we wanted to, we could keep the single `analyticsObserver` and use it in all the state machine instances since it only uses the data it's passed and has no state otherwise. Not sure if that'd be more or less confusing.

Reply:
yeah, i think you're right - trying it out.  simpler is better